burn-core/src/nn/initializer.rs

```diff
@@ -10,74 +10,116 @@ use crate as burn;
 /// Enum specifying with what values a tensor should be initialized
 #[derive(Config, Debug, PartialEq)]
 pub enum Initializer {
-    /// Fills tensor with values drawn uniformly between specified values
-    Uniform(f64, f64),
-    /// Must be implemented by caller. TODO change to NormalizedUniform, fills tensor with values drawn uniformly between -sqrt(1/fan_in) and sqrt(1/fan_in).
-    UniformDefault,
-    /// Fills tensor with values drawn from normal distribution with specified mean and std
-    Normal(f64, f64),
     /// Fills tensor with specified value everywhere
     Constant(f64),
     /// Fills tensor with 1s everywhere
     Ones,
     /// Fills tensor with 0s everywhere
     Zeros,
+    /// Fills tensor with values drawn uniformly between specified values
+    Uniform(f64, f64),
+    /// Fills tensor with values drawn uniformly between -sqrt(1/fan_in) and sqrt(1/fan_in).
+    NormalizedUniform,
+    /// Fills tensor with values drawn from normal distribution with specified mean and std
+    Normal(f64, f64),
     /// Fills tensor with values according to the uniform version of Xavier Glorot initialization described in [Understanding the difficulty of training deep feedforward neural networks](https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf)
     XavierUniform(f64),
     /// Fills tensor with values according to the normal version of Xavier Glorot initialization described in [Understanding the difficulty of training deep feedforward neural networks](https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf)
     XavierNormal(f64),
 }
 
 impl Initializer {
-    pub fn init<B: Backend, const D: usize, S: Into<Shape<D>>>(&self, shape: S) -> Tensor<B, D> {
+    pub fn init<B: Backend, const D: usize, S: Into<Shape<D>>>(
+        &self,
+        shape: S,
+        with_bias: bool,
```

Comment:
I don't think it's a good idea to initialize weight and bias with a single function calls, though I admit the code before was pretty ugly. Often, the bias and the weight don't share the same initializer, so I don't think it sould be the default behavior. 

In fact, it's a bit weird to have an initializer config for a module's weight but none for the bias. Maybe we should replace the `bias` flag with an optional initializer.

```rust
/// Configuration to create a [Linear](Linear) layer.
#[derive(Config, Debug)]
pub struct LinearConfig {
    /// The size of the input features.
    pub d_input: usize,
    /// The size of the output features.
    pub d_output: usize,
    /// The type of function used to initialize the weight matrix
    #[config(default = "Initializer::NormalizedUniform")]
    pub initializer_weight: Initializer,
    /// (Optional) The type of function used to initialize the bias vector.
    /// If no initializer is passed, no bias will be used for the linear transformation.
    #[config(default = "Some(Initializer::NormalizedUniform")]
    pub initializer_bias: Option<Initializer>,
}
```

Then the code becomes prettier:

```rust
let weight = self.initializer_weight.init(shape_weight);
let bias = self.initializer_bias.map(|initializer| initializer.init(shape_bias));
```

If we want to go further, we could replace the `Option` with our own enum and implement init on that enum, but I'm not sure if it's a good idea yet 😅.

Reply:
I was also unsure about having one function call for both weight and bias. The reason I did it is that distribution such as the normalized uniform that was previously used or Xavier depend on their fan in, but if we only have the bias shape it cannot know the fan in because it only has the output size.